src/Analyzers/CSharp/Analyzers/NewLines/ArrowExpressionClausePlacement/ArrowExpressionClausePlacementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        public ArrowExpressionClausePlacementDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ArrowExpressionClausePlacementDiagnosticId,
+                   EnforceOnBuildValues.ArrowExpressionClausePlacement,
+                   CSharpCodeStyleOptions.AllowBlankLineAfterTokenInArrowExpressionClause,
+                   new LocalizableResourceString(
+                       nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_arrow_expression_clause_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxTreeAction(AnalyzeTree);
+
+        private void AnalyzeTree(SyntaxTreeAnalysisContext context)
+        {
+            var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInArrowExpressionClause;
+            if (option.Value)
+                return;
+
+            Recurse(context, option.Notification.Severity, context.Tree.GetRoot(context.CancellationToken));
+        }
+
+        private void Recurse(SyntaxTreeAnalysisContext context, ReportDiagnostic severity, SyntaxNode node)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (node is ArrowExpressionClauseSyntax arrowExpressionClause)
+                ProcessArrowExpressionClause(context, severity, arrowExpressionClause);
+
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.IsNode)
+                    Recurse(context, severity, child.AsNode()!);
+            }
+        }
+
+        private void ProcessArrowExpressionClause(
+            SyntaxTreeAnalysisContext context, ReportDiagnostic severity, ArrowExpressionClauseSyntax arrowExpressionClause)
+        {
+            // get
+            //     => 1 + 2;
+            //
+            // Never looks good.  So we don't process in that case.
+            if (arrowExpressionClause.Parent is AccessorDeclarationSyntax)
+                return;
+
+            // Don't bother analyzing nodes that have syntax errors in them.
+            if (arrowExpressionClause.GetDiagnostics().Any(static d => d.Severity == DiagnosticSeverity.Error))
+                return;
+
+            if (IsOk(arrowExpressionClause.ArrowToken))
+                return;
+
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                this.Descriptor,
+                arrowExpressionClause.ArrowToken.GetLocation(),
+                severity,
+                additionalLocations: null,
+                properties: null));
+
+            return;
+
+            static bool IsOk(SyntaxToken token)
```

Comment:
name could be more descriptive

Reply:
yeah, true.  will change to "IsTokenPlacementCorrect".